src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,151 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+
+    private int size;
+    private T[] elementData = (T[]) new Object[DEFAULT_CAPACITY];
+
     @Override
     public void add(T value) {
+        if (size == elementData.length) {
+            resizeArray();
+        }
 
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of bounds Exception");
+        }
 
+        if (size == elementData.length) {
+            resizeArray();
+        }
+
+        shiftElementsToRight(index);
+        elementData[index] = value;
+        size++;
+    }
+
+    private void resizeArray() {
```

Comment:
private methods should be below public ones

Reply:
Done